src/utils/dateUtils.ts

```diff
@@ -0,0 +1,10 @@
+export function isSameDate(date1?: Date, date2?: Date) {
+  return (
+    date1?.getFullYear() === date2?.getFullYear() &&
+    date1?.getMonth() === date2?.getMonth() &&
+    date1?.getDate() === date2?.getDate()
+  );
+}
+export function isSameHourAndMinute(date1?: Date, date2?: Date) {
```

Comment:
I think `isSameTime` is better, WDYT?

Reply:
I think same hour and time is clearer because the check is different than just checking with `getTime` which also checks for the milliseconds equality.